File: sort-transform.py
"""Given an array of numbers, return a string made up of four parts:

a) a four character 'word', made up of the characters derived from the first two and last two numbers in the array. order should be as read left to right (first, second, second last, last),

b) the same as above, post sorting the array into ascending order,

c) the same as above, post sorting the array into descending order,

d) the same as above, post converting the array into ASCII characters and sorting alphabetically.

The four parts should form a single string, each part separated by a hyphen: '-'

example format of solution: 'asdf-tyui-ujng-wedg'
"""

def sort_transform(arr):
      
    word_list = []

    word1 = transform(arr)

    word2 = transform(sorted(arr))
    
    word3 = transform(sorted(arr, reverse = True))

    arr_ascii = arr_to_ascii(arr)
    ascii_word = [arr_ascii[0], arr_ascii[1], arr_ascii[-2], arr_ascii[-1]]
    word4 = "".join(ascii_word)
    
    word_list.extend([word1, word2, word3, word4])
    
    return "-".join(word_list)
    
def transform(arr):
    numlist = [arr[0], arr[1], arr[-2], arr[-1]]
    word = []
    for num in numlist:
        letter = chr(num)
        word.append(letter)

    return "".join(word)


def arr_to_ascii(arr):
    ascii_list = []

    for item in arr:
        letter = chr(item)
        ascii_list.append(letter)

    ascii_list.sort()

    return ascii_list

File: sort-transform.js
'use strinct';

function sortTransform(arr) {
      
    let wordList = []

    let word1 = transform(arr)

    let arrAscii = arr_to_ascii(arr)
    let asciiWord = [arrAscii[0], arrAscii[1], arrAscii[arrAscii.length-2], arrAscii[arrAscii.length-1]]
    let word4 = asciiWord.join("")

    arr.sort(function(a, b){return a-b});
    let word2 = transform(arr)
    
    arr.reverse();
    let word3 = transform(arr)
    
    let ansList = wordList.concat([word1, word2, word3, word4])
    
    return ansList.join("-")

}
    
function transform(arr) {
    let numlist = [arr[0], arr[1], arr[arr.length-2], arr[arr.length-1]]
    let word = []
    for (let x = 0; x < numlist.length; x++) {
        let letter = String.fromCharCode(numlist[x])
        word.push(letter)
    }

    return word.join("")

}

function arr_to_ascii(arr) {
    let ascii_list = []

    for (let x = 0; x < arr.length; x++) {
        let letter = String.fromCharCode(arr[x])
        ascii_list.push(letter)
    }
    ascii_list.sort();

    return ascii_list

}

console.log(sortTransform([51, 62, 73, 84, 95, 100, 99, 126]))
console.log('3>c~-3>d~-~d>3-3>d~')